### PR TITLE
use namespace to avoid name conflict between tinyxml2 and msxml

### PR DESCRIPTION
**Pull request by johnsonshih**
use namespace to avoid name conflict between tinyxml2 and msxml

### CODE REVIEW

File: src/rospack.cpp [hunk 1]
```diff
@@ -80,7 +80,8 @@
 #define PyUnicode_FromString PyString_FromString
 #endif
 
-using namespace tinyxml2;
+// use namespace to disambiguate from MSXML on Windows
+namespace tx2 = tinyxml2;
```
Comment (dirk-thomas) [inline src/rospack.cpp]: Instead of defining another namespace (`tx2`) please use the original namespace instead (`tinyxml2`).

Comment (johnsonshih) [reply src/rospack.cpp]: sure, change to use namespace tinyxml2

**Comment by dirk-thomas** [comment]
Thank you for the patch.

Out of curiosity why does it conflict on Windows? Is any of the Windows specific headers setting `using namespace msxml`?

**Comment by johnsonshih** [comment]
There is a COM class name XMLDocument in global space from msxml,  that cause name conflict with the tinyxml2::XMLDocument

**Comment by dirk-thomas** [comment]
> There is a COM class name XMLDocument in global space from msxml ...

Thanks for the details. Maybe that should be fixed too :wink:

**Pull request merged**